pkg/server/endpoints/harvester.go

```
@@ -43,13 +47,108 @@ func NewHarvesterAPIHandlers(l logrus.FieldLogger, ds datastore.Datastore, jwtIs
-func (h *HarvesterAPIHandlers) PatchRelationshipsRelationshipID(ctx echo.Context, relationshipID api.UUID) error {
+// PatchRelationship accept/denies relationships requests - (PATCH /relationships/{relationshipID})
+func (h *HarvesterAPIHandlers) PatchRelationship(echoCtx echo.Context, relationshipID api.UUID) error {
+	ctx := echoCtx.Request().Context()
+
+	authTD, ok := echoCtx.Get(authTrustDomainKey).(*entity.TrustDomain)
+	if !ok {
+		err := errors.New("no authenticated trust domain")
+		return h.handleErrorAndLog(err, err.Error(), http.StatusUnauthorized)
+	}
+
+	// get the relationships for the trust domain
+	relationship, err := h.Datastore.FindRelationshipByID(ctx, relationshipID)
+	if err != nil {
+		msg := "error looking up relationships"
+		err := fmt.Errorf("%s: %w", msg, err)
+		return h.handleErrorAndLog(err, msg, http.StatusInternalServerError)
+	}
+
+	if relationship == nil {
+		err := fmt.Errorf("relationship not found")
+		return h.handleErrorAndLog(err, err.Error(), http.StatusNotFound)
+	}
+
+	if relationship.TrustDomainAID != authTD.ID.UUID && relationship.TrustDomainBID != authTD.ID.UUID {
+		err := fmt.Errorf("relationship doesn't belong to the authenticated trust domain")
+		return h.handleErrorAndLog(err, err.Error(), http.StatusUnauthorized)
+	}
+
+	var patchRequest harvester.PatchRelationship
+	if err := chttp.FromBody(echoCtx, &patchRequest); err != nil {
+		msg := "error reading body"
+		err := fmt.Errorf("%s: %w", msg, err)
+		return h.handleErrorAndLog(err, msg, http.StatusBadRequest)
+	}
+
+	consentStatus := patchRequest.ConsentStatus
+
+	switch consentStatus {
+	case api.Accepted, api.Denied, api.Pending:
+	default:
+		err := fmt.Errorf("invalid consent status: %q", consentStatus)
+		return h.handleErrorAndLog(err, err.Error(), http.StatusBadRequest)
+	}
+
+	// update the relationship consent status for the authenticated trust domain
+	if relationship.TrustDomainAID == authTD.ID.UUID {
```

Shouldn't we check explicitly that the relationship.TrustDomainB == authTD.ID.UUID instead of having the else statement? Could we have a case that relationship.TrustdomainA or B does not match authTD.ID.UUID? Should we return an error if the authenticated trust domain does not match A or B in the relationship?

---

There is a validation above preventing that case:

```
if relationship.TrustDomainAID != authTD.ID.UUID && relationship.TrustDomainBID != authTD.ID.UUID {
		err := fmt.Errorf("relationship doesn't belong to the authenticated trust domain")
		return h.handleErrorAndLog(err, err.Error(), http.StatusUnauthorized)
	}
```